ddtrace/tracer/spancontext.go

```diff
@@ -246,11 +266,23 @@ func (t *trace) finishedOne(s *span) {
 	if len(t.spans) != t.finished {
 		return
 	}
-	if tr, ok := internal.GetGlobalTracer().(*tracer); ok {
-		// we have a tracer that can receive completed traces.
-		tr.pushTrace(t.spans)
-		atomic.AddInt64(&tr.spansFinished, int64(len(t.spans)))
+	defer func() {
+		t.spans = nil
+		t.finished = 0 // important, because a buffer can be used for several flushes
+	}()
+	tr, ok := internal.GetGlobalTracer().(*tracer)
+	if !ok {
+		return
+	}
+	// we have a tracer that can receive completed traces.
+	atomic.AddInt64(&tr.spansFinished, int64(len(t.spans)))
+	sd := samplingDecision(atomic.LoadInt64((*int64)(&t.samplingDecision)))
+	if sd == decisionNone {
```

Comment:
This ties `decisionNone` to `p0` which is unfair. What if we just did this instead?
```suggestion
	if p, ok := t.samplingPriority(); ok && p == ext.PriorityAutoReject {
```
You should move this condition below, inside `if sd != decisionKeep`

The reason is: let's not make `decisionNone` be related to priority sampling. Let's just have it mean "undecided".